lib/index.js

```diff
@@ -440,613 +442,602 @@ class ProbePlugin {
+   *
+   * There are two kinds of measures:
+   *  - single document measure, for instance a measure coming from a counter probe
+   *  - measures containing a set of content. These measures generate
+   *      1 document per content. For instance sampler probes generate a statistical
+   *      sample set of documents/messages.
+   *      These measures generates as many measure documents than collected contents.
+   *
+   * @param {Object} probe
+   * @param {Object} measure
+   */
+  _saveMeasure(probe, measure) {
+    const
+      timestamp = Date.now();
+    let
+      promise,
+      request;
 
-    request = [];
+    if (measure.content) {
+      if (measure.content.length === 0) {
+        return Bluebird.resolve(false);
+      }
 
-    measure.content.forEach(content => {
-      request.push({
-        index: {
-          _index: index,
-          _type: probe.name
-        }
-      });
+      request = [];
 
-      request.push({
-        timestamp,
-        content
-      });
-    });
+      for (const content of measure.content) {
+        request.push({
+          index: {
+            _index: this.index,
+            _type: probe.name
+          }
+        });
 
-    promise = context.accessors.execute(context.constructors.Request({
-      index,
-      collection: probe.name,
-      controller: 'bulk',
-      action: 'import',
-      body: {
-        bulkData: request
+        request.push({
+          timestamp,
+          content
+        });
       }
-    }));
-  }
-  else {
-    request = measure;
-    request.timestamp = timestamp;
 
-    try {
-      promise = context.accessors.execute(context.constructors.Request({
-        index,
+      promise = this.context.accessors.execute(this.context.constructors.Request({
+        index: this.index,
         collection: probe.name,
-        controller: 'document',
-        action: 'create',
-        body: JSON.parse(JSON.stringify(request))
-      }, {
-        user: {
-          _id: null
+        controller: 'bulk',
+        action: 'import',
+        body: {
+          bulkData: request
         }
       }));
     }
-    catch (error) {
-      promise = Bluebird.reject(error);
+    else {
+      request = measure;
+      request.timestamp = timestamp;
+
+      try {
+        promise = this.context.accessors.execute(this.context.constructors.Request({
+          index: this.index,
+          collection: probe.name,
+          controller: 'document',
+          action: 'create',
+          body: JSON.parse(JSON.stringify(request))
+        }, {
+          user: {
+            _id: null
+          }
+        }));
+      }
+      catch (error) {
+        promise = Bluebird.reject(error);
+      }
     }
+
+    return promise;
   }
 
-  return promise;
-}
+  /**
+   * Performs a set of operations after receiving a probe.
+   *
+   * @param {Object} probe
+   * @param {Object} measure
+   */
+  _afterMeasure(probe, measure) {
+    return new Promise((resolve, reject) => {
```

Comment:
should be `return new Bluebird((resolve, reject) => {`

Reply:
indeed
Done.